### ISSUE TITLE
validation on fragment is not working after back pressed

### ISSUE THREAD

**Issue by ParitoshVaidya** [wellingtoncosta/convalida#50]
I am using kotlin DSL for validation. I have multiple fragments opening after one another for user input. Each fragment has validation using code:

**Sample code**

	private val validations by lazy {
		validationSet(
			validations = listOf(
				binding.txtSomething.isRequired(errorMessage = getString(R.string.field_required))
			),
			actions = actions
					validateByClickingOn binding.btnNext
					whenOnSuccess ::onValidationSuccess
		)
	}

If I press the back button in between, validation stops working.

**Observations**:
1. `apply { validations }` gets called once again once user comes back on fragment
2.  `validateByClickingOn binding.btnNext`
`whenOnSuccess ::onValidationSucces` does not get called second time

**Comment by wellingtoncosta** [comment]
Hi @ParitoshVaidya 

Firstly thank you for using Convalida.

I would like to recommend you to try the new (but currently in SNAPSHOT) Kotlin DSL structure. It's more concise, more flexible and more intuitive.

First, you need to add the Sonatype's snapshots repository:

`https://oss.sonatype.org/content/repositories/snapshots`

Gradle Groovy DSL:

```groovy
maven {
    url = 'https://oss.sonatype.org/content/repositories/snapshots'
}
```

Gradle Kotlin DSL:

```kotlin
maven(url = "https://oss.sonatype.org/content/repositories/snapshots")
```

And then, add the SNAPSHOT artifacts:

If you just use the Convalida Kotlin DSL:

`io.github.wellingtoncosta:convalida-ktx:3.2.0-SNAPSHOT`

If you also use Convalida Annotations and/or Data Binding support:

`io.github.wellingtoncosta:convalida-runtime:3.2.0-SNAPSHOT`
`io.github.wellingtoncosta:convalida-compiler:3.2.0-SNAPSHOT`

The Convalida website isn't updated yet with new features, but you can take a look at the `sample` module. Specifically, you could see this: 

https://github.com/wellingtoncosta/convalida/blob/develop/sample/src/main/java/convalida/sample/KotlinDslSampleActivity.kt

I need to take a time to finish some unfinished features and so release `3.2.0`. I should have done it a few weeks ago.

**Comment by ParitoshVaidya** [comment]
I tried:

	convalida {
		field(binding.txtFirstName) {
			isRequired(errorMessage = fieldRequired)
		}
		field(binding.txtLastName) {
			isRequired(errorMessage = fieldRequired)
		}
		validateOn(binding.btnNext) {
			onSuccess { onValidationSuccess() }
			onError { onValidationError() }
		}
	}

`onError` is executing correctly. `onValidationSuccess()` is not getting called.

**Comment by wellingtoncosta** [comment]
Ok.. It's so stranger. I'll see it ASAP.

Can you send me repo link? (if it's public)

**Comment by ParitoshVaidya** [comment]
Unfortunately, my repo is not public. We can test this with 2 sample fragments though.

The scenario is:

1. if validation fails, it can not succeed next time.
2. if validation is successful, onValidationSuccess() gets called correctly.

**Comment by wellingtoncosta** [comment]
Ok.

Free feel to implement an example to facilitate the investigation. When we fix this bug, it could be added in `sample` module.

**Comment by ParitoshVaidya** [comment]
[Link to repository](https://github.com/ParitoshVaidya/ConvalidaValidationSample)

Kill the app and start over to verify each scenario:

1. if validation fails, it can not succeed even if user enters text in EditText field.
2. if validation is successful, onValidationSuccess() gets called correctly.

**Comment by wellingtoncosta** [comment]
Hi @ParitoshVaidya 

I found the bug. It was a wrong logic that check if has any field invalid, but I'm was not setting true when has not any invalid field.

I'm publishing a snapshot update ASAP.

Thanks for reporting the bug! :)

**Comment by wellingtoncosta** [comment]
Hi @ParitoshVaidya 

After a lot of problems with my environment, I published the 3.2.0-SNAPSHOT update.

Please try again and let me know.

I hope it works. :)

**Comment by ParitoshVaidya** [comment]
Fixed
